forum/users/forms.py

```diff
@@ -0,0 +1,14 @@
+from django import forms
+from .models import Role
+from django.contrib.auth.forms import AuthenticationForm
+
+class UserLoginForm(AuthenticationForm):
+    username = forms.EmailField(label="Email")
+    password = forms.CharField(widget=forms.PasswordInput, label="Password")
+
+class RoleSelectionForm(forms.Form):
+    role = forms.ModelChoiceField(
+        queryset=Role.objects.all(), 
+        empty_label="Select Role", 
+        widget=forms.Select(attrs={'class': 'form-control'})
+    )
```

Comment:
If no roles are available in the system, users may see an empty dropdown in the `RoleSelectionForm`. To handle this case gracefully, you could display a message instead of an empty form field. This can be done in the view or by raising a validation error if there are no roles.
```
class RoleSelectionForm(forms.Form):
    role = forms.ModelChoiceField(
        queryset=Role.objects.all(), 
        empty_label="Select Role", 
        widget=forms.Select(attrs={'class': 'form-control'})
    )
    
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        if not Role.objects.exists():
            self.fields['role'].widget.attrs['disabled'] = 'disabled'
            self.fields['role'].help_text = "No roles available. Please contact support."
```